src/forms/currentIncome.js

```diff
@@ -60,8 +59,11 @@ const IncomeForm = function ({ storeComplex, client }) {
   };  // End ensureFuture()
 
 
-  var sharedProps = { client: client, type: type, time: time,
-                      storeComplex: ensureFuture };
+  var sharedProps = {
```

Comment:
es6 has property value short hand   :) 

```
let sharedProps = {
   client: client,
   time: time,
   type: type
}
```
Do the same thing 

```
let sharedProps = {
   client,
   time,
   type
}
```
cool

not required but I suggest moving 'type: type' on its own line.

Reply:
Which ES6 features to use is definitely a discussion we should have. For now, I'll just put them on separate lines.